### PR TITLE
Upgrade compiler and tcgc in Aug

### LINKED ISSUE #2740: Upgrade compiler version to v0.59.0
Upgrade compiler version to `v0.59.0`

### PR DESCRIPTION
**Pull request by v-jiaodi**
fixes #2740 
Upgrade compiler version to `v0.59.0`
Upgrade tcgc version to `v0.45.1`

### CODE REVIEW

File: packages/typespec-test/test/todo_non_branded/generated/openapi/openapi.json [hunk 1]
```diff
@@ -90,6 +90,7 @@
           }
         },
         "requestBody": {
+          "required": true,
```
Comment (MaryGao) [inline packages/typespec-test/test/todo_non_branded/generated/openapi/openapi.json:93]: Should be a fix in OpenAPI emitter with shared route: https://github.com/microsoft/typespec/pull/4044

File: packages/typespec-test/test/openai_generic/generated/typespec-ts/src/models/models.ts [hunk 2]
```diff
@@ -501,7 +501,7 @@ export interface Embedding {
   object: "embedding";
   /**
    * The embedding vector, which is a list of floats. The length of vector depends on the model as\
-   *    * listed in the [embedding guide](/docs/guides/embeddings).
+   * listed in the [embedding guide](/docs/guides/embeddings).
```
Comment (MaryGao) [inline packages/typespec-test/test/openai_generic/generated/typespec-ts/src/models/models.ts:504]: The change should be expected and fixed here: https://github.com/microsoft/typespec/issues/3646.


which means the nightly ci would be expected to be failed.

**Review by qiaozha — APPROVED** [review]

File: common/config/rush/pnpm-lock.yaml [hunk 3]
```diff
@@ -1,43 +1,44 @@
 lockfileVersion: 5.4
 
 importers:
+
```
Comment (qiaozha) [inline common/config/rush/pnpm-lock.yaml:4]: why do we have 7927 lines changes for this file ?

Comment (MaryGao) [reply common/config/rush/pnpm-lock.yaml:4]: i think this is because of format.

**Pull request merged**